wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java

```diff
@@ -1010,21 +1010,37 @@ private void cleanup() {
 
         executor.shutdown();
 
+        boolean interrupted = false;
+        final ExecutorService destroyMachinesExecutor = Executors.newCachedThreadPool(
```

Comment:
It is logically calculate termination timeout using number of machines, but note, that task is terminate it all in one minute! Also different machines may have different destroying time.